src/Raven.Server/Documents/Indexes/Sharding/Persistence/Corax/ShardedCoraxIndexReadOperation.cs

```diff
@@ -65,7 +67,19 @@ private ShardedQueryResultDocument AddOrderByFields(Document queryResult, IndexQ
                 throw new NotSupportedInShardingException("Ordering by score is not supported in sharding");
             }
 
-            var orderByFieldMetadata = orderByFields[i];
+            var orderByFieldMetadata = orderByFields[currentCoraxOrderIndex];
+            
+            if (orderByFieldMetadata.Field.FieldName.ToString() != orderByField.Name)
```

Comment:
Are we materializing the string on each call here?

Reply:
Yes, `orderByFieldMetadata.Field.FieldName` is Voron slice, I'll change it in next commit

Reply:
Now we're comparing slices instead